src/coreEnforcer.ts

```diff
@@ -470,6 +470,7 @@ export class CoreEnforcer {
 
         const eft = parameters['p_eft'];
         if (eft && eftRes === Effect.Allow) {
+          explainIndex = i;
```

Comment:
This code is still incorrect, please remove this.

The follow is my design: 

You need to change the `explainIndex`, which should be an array, then change the `EffectorStream` interface like:
```ts
export interface EffectorStream {
  current(): boolean;

  pushEffect(eft: Effect): [boolean, boolean] | [boolean, boolean, boolean];
}
```

This change is fully compatible with the new and old library.

`[boolean, boolean, boolean]`:
1. [0] means the enforced result
2. [1] means whether to record this rule, we need to recode this rule in enforce method when this is true
3. [3] means whether to continue the enforce


I didn't verify this design, could you help me verify this?

Reply:
@nodece so do we also put all the possible reasons for deny and allow in explainIndex, because then we would have to go till the end of the policy.

Reply:
> 

When [3] is true, we will break this loop.

Reply:
okay so we are keeping it the same and breaking at deny, but we do keep adding all allow policies right?

Reply:
> okay so we are keeping it the same and breaking at deny, but we do keep adding all allow policies right?

For `some(where (p_eft == allow)) && !some(where (p_eft == deny))`

If no deny rule, we should return one allow rule, otherwise, we return one deny rule.

 > I'm just trying to understand why we are keeping the `explainIndex` an array

Sorry, the old type is correct.

Reply:
@nodece I have added it like it will always record the reason and return it